--- a/modules/cudaarithm/CMakeLists.txt
+++ b/modules/cudaarithm/CMakeLists.txt
@@ -12,11 +12,11 @@
   if(UNIX AND NOT BUILD_SHARED_LIBS AND CUDA_VERSION_STRING VERSION_GREATER_EQUAL 9.2 AND CMAKE_VERSION VERSION_GREATER_EQUAL 3.23)
     set(CUDA_FFT_LIB_EXT "_static_nocallback")
   endif()
-  list(APPEND extra_dependencies CUDA::cudart_static CUDA::nppial${CUDA_LIB_EXT} CUDA::nppc${CUDA_LIB_EXT} CUDA::nppitc${CUDA_LIB_EXT} CUDA::nppig${CUDA_LIB_EXT} CUDA::nppist${CUDA_LIB_EXT} CUDA::nppidei${CUDA_LIB_EXT})
+  list(APPEND extra_dependencies CUDA::cudart_static CUDA::nppial CUDA::nppc CUDA::nppitc CUDA::nppig CUDA::nppist CUDA::nppidei)
   if(HAVE_CUBLAS)
-    list(APPEND optional_dependencies CUDA::cublas${CUDA_LIB_EXT})
+    list(APPEND optional_dependencies CUDA::cublas)
     if(NOT CUDA_VERSION VERSION_LESS 10.1)
-      list(APPEND optional_dependencies CUDA::cublasLt${CUDA_LIB_EXT})
+      list(APPEND optional_dependencies CUDA::cublasLt)
     endif()
   endif()
   if(HAVE_CUFFT)
